kernel/src/main/java/org/kframework/parser/ParserUtils.java

```diff
@@ -220,12 +220,20 @@ public Set<Module> loadModules(
 
         int errors = 0;
         for (String moduleName : duplicateModules) {
-          Tuple3<String, Source, Location> firstMod = groupedModules.get(moduleName).get(0);
-          Tuple3<String, Source, Location> secondMod = groupedModules.get(moduleName).get(1);
-          KEMException ex = KEMException.outerParserError("Module " + moduleName + " previously declared at "
-                  + firstMod._2() + " and " + firstMod._3(), secondMod._2(), secondMod._3());
-          errors++;
-          kem.addKException(ex.getKException());
+          Tuple4<String, Source, Location, String> firstMod = groupedModules.get(moduleName).get(0);
+          Tuple4<String, Source, Location, String> secondMod = groupedModules.get(moduleName).get(1);
+          // give an error message only if we have
+          // the same module name found in different filenames (path doesn't matter)
+          // the location is different or
+          // the hashes of the pretty printed contents are different.
+          if (!Paths.get(firstMod._2().source()).getFileName().equals(Paths.get(secondMod._2().source()).getFileName())
+            || !firstMod._3().equals(secondMod._3())
+            || !firstMod._4().equals(secondMod._4())) {
+              KEMException ex = KEMException.outerParserError("Module " + moduleName + " differs from previous declaration at "
+                      + firstMod._2() + " and " + firstMod._3(), secondMod._2(), secondMod._3());
+              errors++;
+              kem.addKException(ex.getKException());
+          }
```

Comment:
> I don't think the error message in the test here is particularly clear - the modules here **are** in fact identical, but happen to be declared in the same file. We probably want to distinguish (if possible) this case from "the exact same module, but moved on disk".

@Baltoli do you have a suggestion?
I can't really think of a better message with what I have in hand:
- file name (trying to exclude the path)
- location in the file
- hashed contents

Reply:
Ah, good point. In that case I'd leave it as you have it in this change, and if we come up to a particular case that confuses someone we can revisit it.